.github/BEST_PRACTICE.md

```diff
@@ -0,0 +1,171 @@
+The conventions outlined below should be adhered to in all contributions to Kindly.
+
+This document assumes you have reviewed the [Contributing guidance](CONTRIBUTING.md) already. If you haven't, please take a moment to do so before continuing.
+
+<hr>
+
+# Naming Conventions
+
+- for **Components**:
+  - use PascalCase
+- for **Types**:
+  - use PascalCase
+  - suffix it with the word 'Type'
+    - e.g.: MessageType
+  - always declare **Type**, not _Interface_
+- for **Prop Types**:
+  - use PascalCase
+  - prefix it with the relevant Component name
+  - suffix it with the word 'Props'
+    - e.g.: MessageCardProps
+- for **Functions**:
+  - use camelCase
+- for **Branches**:
+  - prefix it with the number of the issue it relates to
+  - suffix it with a descriptive line on the work being done
+  - separate the issue number from the description with a forward slash `/`
+  - hyphenate the description `-`
+    - e.g.: #273/create-new-component
+
+### Never Abrev ❗️
+
+<hr>
+
+# Types
+
+- When Types are declared within a component file:
+  - They should not be exported
+  - They should be declared at the top of the file
+- When Types are declared in a **\***Types.ts\*\*\* file:
+  - They should be relevant to the scope of the feature the file relates to
+    - e.g.: if a Type is declared in "conversationsTypes.ts" it should be relevant to several files connected with the '/conversations' feature
+    - if a Type declared in "conversationsTypes.ts" is only exported into one file, it should be moved there
+
+## Avoid Type ambiguity:
+
+Types should not be declared with Logical OR Operators `|` when possible. For example:
+
+**This:**
+
+```ts
+const [isTrue, setIsTrue] = useState<boolean>(1 + 1 === 2);
+// a "relevant expression" evaluates to true and iniates the state truthfully
+```
+
```

Comment:
I'm not sure about this at all. I think it's correct to sometime initiate a value to null and deal with that in the code is the correct way to use hooks. For example useContext and useRef both initiate to null and then you deal with that later in the code to provide meaningful error messages when the value is null. Like outside a context provider.

Reply:
Following our chat earlier I'm resolving this one - we'll approach this with some reference to already existing guidance and monitor cases individually in PR reviews